src/test/java/com/rocketden/main/util/UtilityTests.java

```diff
@@ -30,7 +36,7 @@ public void generateValidRoomId() {
     @Test
     public void generateValidUserId() {
         // Verify user ids are generated correctly
-        String userId = utility.generateId(UserService.USER_ID_LENGTH);
+        String userId = utility.generateUniqueId(UserService.USER_ID_LENGTH, Utility.USER_ID_KEY);
 
         assertEquals(UserService.USER_ID_LENGTH, userId.length());
 
```

Comment:
Suggestion: Add a test that mocks the repository to return true when findRoomById is called the first time, and check that it generates a second id

Reply:
I added a test for this, but that required that the `utility` I used is a `@Spy` with `@InjectMocks` so I could inject the `RoomRepository` and `UserRepository` mocks.

On several methods I did not include output for those mocks, and I believe that they default to `null`, which is the desired behavior on these cases. Please let me know if this is correct, and I can also add a comment explaining this (if it is true).

Reply:
Yep I think you're right about the return null so it should be good :)